awscli/customizations/eks/update_kubeconfig.py

```diff
@@ -235,14 +236,25 @@ def _expand_path(self, path):
 
 
 class EKSClient(object):
-    def __init__(self, session, cluster_name, role_arn, parsed_globals=None):
+    # A set that contains optional flag keys here. 
+    # If you need to specify a default value other than `None` for the parsed_args,
+    # you need to provide a value for the 'default' key in ARG_TABLE. 
+    _OPTIONAL_ATTRS = {
+        'role_arn',
+    }
+
+    def __init__(self, session, *, cluster_name, parsed_globals=None, **parsed_args):
         self._session = session
-        self._cluster_name = cluster_name
-        self._role_arn = role_arn
         self._cluster_description = None
+        self._cluster_name = cluster_name
         self._globals = parsed_globals
+        
+        for name in EKSClient._OPTIONAL_ATTRS:
+            setattr(self, f'_{name}', parsed_args[name])
 
-    def _get_cluster_description(self):
+    @property
+    @lru_cache()
```

Comment:
Do we need to use an `lru_cache` here? It looks like the method already has it's own caching implementation.